nodes/wot-discovery.js

```diff
@@ -147,6 +159,39 @@ module.exports = function (RED) {
 
             return addresses;
         }
+        function _getDiscoveryLinks(){
+            if(config.useCoap){
+                coapAddresses.forEach((address)=>{
+                    var reqOpt = url.parse(`coap://${address}/.well-known/core`);
+                    reqOpt.pathname = reqOpt.path;
+                    reqOpt.query =`rt=wot.thing`;
+                    reqOpt.multicast = true;
+                    var req = coap.request(reqOpt);
+                    req.on("response", _coreResponse);
+                    req.on("error", function (err) {
+                        node.log("client error");
+                        node.log(err);
+                    });
+                    req.end();
+                });
+            }
+        }
+
+        function _coreResponse(res){
+            res.on("data", (data) => {
+                if(res.headers["Content-Format"]==="application/link-format"){
+                    var links=data.toString().match(/<.*?>/g);
+                    for(var i=0; i<links.length; i++){
+                        links[i]=links[i].replace('<', '');
+                        links[i]=links[i].replace('>', '');
+                        node.log(links[i]);
+                        coapAddresses.forEach((address)=>{
+                            _sendCoapDiscovery(address, links[i]);
+                        });
+                    }
+                }
+            });
+        }
```

Comment:
The following is a rather complex suggestion which does both some refactoring and adds filter capabilities to the function. I already tested it with an ESP with WoT support and it worked :)

Maybe you could have a look on the TODO? A regEx assertion like you did before could be nice to ensure that the first parameter has the appropriate link format.

One thing we could discuss (and also refactor) is the actual filtering by resource and content type. In theory a resource could have multiple resource types which is why I was wondering if we should also convert the args to an array, iterate over them, and check if `"wot.thing"` is one of them. However, I guess a thing description should only have this one resource type which should make this additional step unnecessary. 

```suggestion
                 if (res.headers["Content-Format"] === "application/link-format") {
                    let links = data.toString().split(',');
                    
                    links = links.map(link => {
                        return link.split(";");
                    });
                    
                    links.forEach(link => {

                        let correctResourceType = false;
                        let correctContentType = false;
                        let path;

                        link.forEach(function (curentValue, index) {
                            if (index === 0) {
                                // First parameter must be the path
                                // TODO: Add assertion for </ ... > format
                                path = curentValue.substring(2, curentValue.length - 1);
                                return;
                            } else if (!path) {
                                return;
                            }

                            curentValue = curentValue.split("=");
                            let parameter = curentValue[0];
                            let values = curentValue[1]

                            switch (parameter) {
                                case "ct":
                                    if (values === "432") {
                                        correctContentType = true;
                                    }
                                    break;
                                case "rt":
                                    if (values === '"wot.thing"') {
                                        correctResourceType = true;
                                    }
                                    break;
                            }
                        })

                        if (correctContentType && correctResourceType) {
                            coapAddresses.forEach((address) => {
                                _sendCoapDiscovery(address, path);
                            });
                        }
                    });
                }
```